src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -77,12 +77,13 @@ ScheduleMap GcsPackStrategy::Schedule(
   // First fill up a node. If the node resource is insufficient, select a new node.
   // TODO(ffbin): We will speed this up in next PR. Currently it is a double for loop.
   ScheduleMap schedule_map;
-  const auto &alive_nodes = context->node_manager_.GetClusterRealtimeResources();
+  absl::flat_hash_map<NodeID, ResourceSet> alive_nodes(
+      context->node_manager_.GetClusterRealtimeResources());
```

Comment:
I have a problem here, if current cluster resource can't satisfied the required resource, return the `schedule_map` directly, that's right, but if it can satisfied the required resource, should we substract the required resource from the `bundle->GetRequiredResources()` reference before return the `schedule_map` ? If not, It maybe cause many other pg schedule successful, but create failed, because heartbeat interval is far greater than pull a pg from local memory queue. 
cc @rkooo567 @ffbin

Reply:
No, scheduler should never change the resource view.

Reply:
This issue can be solved by this approach; https://github.com/ray-project/ray/pull/11894#issuecomment-725261052. This will materialize the resource view while it doesn't modify the original resource view.